packages/react/src/components/Layout/Layout.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { settings as ddsSettings } from '@carbon/ibmdotcom-utilities';
+import { settings } from 'carbon-components';
+import classnames from 'classnames';
+
+const { stablePrefix } = ddsSettings;
+const { prefix } = settings;
+
+/**
+ * Layout types
+ *
+ * @type {{'1-3': string[]}}
+ * @private
+ */
+const _types = {
+  '1-3': [`${prefix}--col-lg-4`, `${prefix}--col-lg-12`], // 1/4 - 3/4
+};
+
+/**
+ * Returns the child classes with the proper column class names
+ *
+ * @param {string} type layout type
+ * @param {object} children child nodes
+ * @returns {*} modified child content
+ * @private
+ */
+function _updateChild(type, children) {
+  const final = [];
+
+  children.map((child, i) => {
+    if (child.props['data-sticky'] === 'true') {
+      final.push(
+        <div
+          className={
+            _types[type] && _types[type][i] ? _types[type][i] : `${prefix}--col`
+          }>
+          {React.cloneElement(child, {
+            className: classnames(
+              child.props.className,
+              `${prefix}--layout--sticky`
+            ),
+            key: i,
+          })}
+        </div>
+      );
+    } else {
+      final.push(
+        React.cloneElement(child, {
+          className: classnames(
+            child.props.className,
+            _types[type] && _types[type][i] ? _types[type][i] : `${prefix}--col`
+          ),
+          key: i,
+        })
+      );
+    }
+  });
+
+  return final;
+}
+
+/**
+ * Returns the vertical rhythm modifier class
+ *
+ * @param {string} position top or bottom position of the layout
+ * @param {string} modifier layout modifier from the layout scale
+ * @returns {*|string} vertical rhythm class
+ * @private
+ */
+function _vrClass(position, modifier) {
+  return modifier && `${prefix}--layout__${position}-${modifier}`;
+}
+
+/**
+ * Layout component
+ * This is an abstract layout component that can be reutilized for multiple
+ * patterns. This autocreates the proper grid classes, along with the optional
+ * sticky attribute to the child elements (columns).
+ *
+ * @param {object} props props object
+ * @param {string} props.type layout type (1-3)
+ * @param {string=} props.marginTop top margin layout class (layout-01 - layout-07)
+ * @param {string=} props.marginBottom top margin layout class (layout-01 - layout-07)
+ * @returns {*} Layout component
+ */
+const Layout = ({ type, marginTop, marginBottom, children }) => (
+  <section
+    data-autoid={`${stablePrefix}--layout`}
+    className={classnames(
+      `${prefix}--grid`,
+      _vrClass('top', marginTop),
+      _vrClass('bottom', marginBottom)
```

Comment:
What I like about this approach is that it locks in with the layout scale. It also reduces an additional dom element added on the page with a wrapper for this. 

The thought pattern for this was also very similar to how you can create `parsys` components in Adobe Experience Manager, where it's a flexible layout component you can drop on the page with then additional child parsys elements that can be dropped in the corresponding columns. But I'm not as familiar with the Drupal side of things so @sannminwin can probably speak more to this. But in this case, the spacing modifiers can be set even from the authoring side. But taking a step back, not saying this should or shouldn't be ultimately used in this way, but at least for pattern creation this can be set on that level and not by authors.

Reply:
I think that was the intent behind the tokens themselves. Carbon only allows so many spacing options to be applied. If Drupal wants to add this type of flexibility (which I don't think they should even necessarily do) then they should use carbon's tokens. It feels like this is breaking the separation of concerns a little. Again, maybe I'm struggling with the scope of this pull request though.

Reply:
I personally think it's a useful addition to this component. I'll let others weigh in though:
@annawen1 @sannminwin @kennylam @andysherman2121 @wonilsuhibm @larahanlon2 @oliviaflory